File: data/recomposed_projects/deepseek-coder-33b-instruct/body/commons-validator/evosuite-test/src/test/org/apache/commons/validator/routines/checkdigit/ModulusTenCheckDigit_ESTest.py
from __future__ import annotations
import time
import re
import os
import typing
from typing import *
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit import *

# from src.test.org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class ModulusTenCheckDigit_ESTest(unittest.TestCase):

    def test18(self) -> None:

        intArray0 = [0, 0, 0, 0]
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        string0 = modulusTenCheckDigit0.toString()
        self.assertEqual(
            "ModulusTenCheckDigit[postitionWeight=[0, 0, 0, 0], useRightPos=False, sumWeightedDigits=False]",
            string0,
        )

    def test17(self) -> None:

        intArray0 = [0] * 17
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)
        boolean0 = modulusTenCheckDigit0.isValid(None)
        self.assertFalse(boolean0)

    def test16(self) -> None:

        intArray0 = [0]
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        boolean0 = modulusTenCheckDigit0.isValid("")
        self.assertFalse(boolean0)

    def test15(self) -> None:

        intArray0 = [0] * 5
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        boolean0 = modulusTenCheckDigit0.isValid("11L'u%")
        self.assertFalse(boolean0)

    def test14(self) -> None:

        intArray0 = [0] * 10
        modulusTenCheckDigit0 = ModulusTenCheckDigit(intArray0, True, True)
        boolean0 = modulusTenCheckDigit0.isValid("37")
        self.assertFalse(boolean0)

    def test13(self) -> None:

        modulusTenCheckDigit0 = None
        try:
            modulusTenCheckDigit0 = ModulusTenCheckDigit(None, True, True)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            # no message in exception (getMessage() returned null)
            # verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e)
            self.assertIsInstance(e, TypeError)

    def test12(self) -> None:

        with pytest.raises(RuntimeError):
            ModulusTenCheckDigit.ModulusTenCheckDigit1(None, True)

    def test11(self) -> None:

        with pytest.raises(RuntimeError):
            ModulusTenCheckDigit.ModulusTenCheckDigit2(None)

    def test10(self) -> None:

        intArray0 = []
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)

        try:
            modulusTenCheckDigit0.isValid("eH;x%=j7qBnzBs7")
            self.fail("Expecting exception: ArithmeticException")

        except ArithmeticError as e:
            self.verifyException(
                "org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit",
                e,
            )

    def test09(self) -> None:

        intArray0 = [0] * 4
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        try:
            modulusTenCheckDigit0._toInt("~", 118, -1)
            self.fail("Expecting exception: Exception")

        except Exception as e:
            #
            # Invalid Character[118] = '~'
            #
            self.assertIsInstance(e, CheckDigitException.CheckDigitException)

    def test08(self) -> None:

        intArray0 = []
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)

        with pytest.raises(ArithmeticException):
            modulusTenCheckDigit0.weightedValue(33, 33, 33)

    def test07(self) -> None:

        intArray0 = [0] * 10
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)

        with self.assertRaises(ArrayIndexOutOfBoundsException):
            modulusTenCheckDigit0.weightedValue(-3218, -3218, -3218)

    def test06(self) -> None:

        intArray0 = [0] * 9
        intArray0[0] = 133
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)
        boolean0 = modulusTenCheckDigit0.isValid("k1")
        self.assertTrue(boolean0)

    def test05(self) -> None:

        intArray0 = [0] * 4
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        int0 = modulusTenCheckDigit0._toInt("r", 1, 118)
        self.assertEqual(27, int0)

    def test04(self) -> None:

        intArray0 = [0] * 13
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0)
        int0 = modulusTenCheckDigit0._toInt("0", 10, -600)
        self.assertEqual(0, int0)

    def test03(self) -> None:

        intArray0 = [0] * 7
        intArray0[1] = -446
        modulusTenCheckDigit0 = ModulusTenCheckDigit(intArray0, True, False)
        int0 = modulusTenCheckDigit0._weightedValue(1871, 4315, 1871)
        self.assertEqual(-834466, int0)

    def test02(self) -> None:

        intArray0 = [1]
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        int0 = modulusTenCheckDigit0._weightedValue(1, 1, 1599)
        self.assertEqual(1, int0)

    def test01(self) -> None:

        intArray0 = [0] * 4
        modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(
            intArray0, False
        )
        int0 = modulusTenCheckDigit0._weightedValue(3301, 701, 9)
        self.assertEqual(0, int0)

    def test00(self) -> None:

        intArray0 = [0] * 8
        intArray0[0] = -4465
        modulusTenCheckDigit0 = ModulusTenCheckDigit(intArray0, True, True)
        int0 = modulusTenCheckDigit0._weightedValue(21, -1538, 1465)
        self.assertEqual(0, int0)

File: java_projects/cleaned_final_projects_evosuite/commons-validator/src/test/java/org/apache/commons/validator/routines/checkdigit/ModulusTenCheckDigit_ESTest.java


/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */



package org.apache.commons.validator.routines.checkdigit;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class ModulusTenCheckDigit_ESTest extends ModulusTenCheckDigit_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      int[] intArray0 = new int[8];
      intArray0[0] = (-4465);
      ModulusTenCheckDigit modulusTenCheckDigit0 = new ModulusTenCheckDigit(intArray0, true, true);
      int int0 = modulusTenCheckDigit0.weightedValue(21, (-1538), 1465);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      int[] intArray0 = new int[4];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      int int0 = modulusTenCheckDigit0.weightedValue(3301, 701, 9);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      int[] intArray0 = new int[1];
      intArray0[0] = 1;
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      int int0 = modulusTenCheckDigit0.weightedValue(1, 1, 1599);
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      int[] intArray0 = new int[7];
      intArray0[1] = (-446);
      ModulusTenCheckDigit modulusTenCheckDigit0 = new ModulusTenCheckDigit(intArray0, true, false);
      int int0 = modulusTenCheckDigit0.weightedValue(1871, 4315, 1871);
      assertEquals((-834466), int0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      int[] intArray0 = new int[13];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      int int0 = modulusTenCheckDigit0.toInt('0', 10, (-600));
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      int[] intArray0 = new int[4];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      int int0 = modulusTenCheckDigit0.toInt('r', 1, 118);
      assertEquals(27, int0);
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      int[] intArray0 = new int[9];
      intArray0[0] = 133;
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      boolean boolean0 = modulusTenCheckDigit0.isValid("k1");
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      int[] intArray0 = new int[10];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      // Undeclared exception!
      try { 
        modulusTenCheckDigit0.weightedValue((-3218), (-3218), (-3218));
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index -9 out of bounds for length 10
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      int[] intArray0 = new int[0];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      // Undeclared exception!
      try { 
        modulusTenCheckDigit0.weightedValue(33, 33, 33);
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // / by zero
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      int[] intArray0 = new int[4];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      try { 
        modulusTenCheckDigit0.toInt('~', 118, (-1));
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Invalid Character[118] = '~'
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.CheckDigitException", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      int[] intArray0 = new int[0];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      // Undeclared exception!
      try { 
        modulusTenCheckDigit0.isValid("eH;x%=j7qBnzBs7");
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // / by zero
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      // Undeclared exception!
      try { 
        ModulusTenCheckDigit.ModulusTenCheckDigit2((int[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      // Undeclared exception!
      try { 
        ModulusTenCheckDigit.ModulusTenCheckDigit1((int[]) null, true);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      ModulusTenCheckDigit modulusTenCheckDigit0 = null;
      try {
        modulusTenCheckDigit0 = new ModulusTenCheckDigit((int[]) null, true, true);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.validator.routines.checkdigit.ModulusTenCheckDigit", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      int[] intArray0 = new int[10];
      ModulusTenCheckDigit modulusTenCheckDigit0 = new ModulusTenCheckDigit(intArray0, true, true);
      boolean boolean0 = modulusTenCheckDigit0.isValid("37");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      int[] intArray0 = new int[5];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      boolean boolean0 = modulusTenCheckDigit0.isValid("11L'u%");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      int[] intArray0 = new int[1];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      boolean boolean0 = modulusTenCheckDigit0.isValid("");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      int[] intArray0 = new int[17];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit2(intArray0);
      boolean boolean0 = modulusTenCheckDigit0.isValid((String) null);
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      int[] intArray0 = new int[4];
      ModulusTenCheckDigit modulusTenCheckDigit0 = ModulusTenCheckDigit.ModulusTenCheckDigit1(intArray0, false);
      String string0 = modulusTenCheckDigit0.toString();
      assertEquals("ModulusTenCheckDigit[postitionWeight=[0, 0, 0, 0], useRightPos=false, sumWeightedDigits=false]", string0);
  }
}
